src/hooks/useArticle.js

```diff
@@ -0,0 +1,54 @@
+import { useEffect, useState } from "react";
+import { apiClient } from "../../api";
+
+// Load the article and the comments
+// increment the vote count <-- with optimistic updates
+// create a new comment <-- with optimistic updates?
+
+export const useArticle = (id) => {
+  const [article, setArticle] = useState(null);
+  const [error, setError] = useState(null);
+
+  useEffect(() => {
+    loadArticle(id)
+      .then((res) => {
+        setArticle(res);
+      })
+      .catch((err) => {
+        setError(err);
+      });
+  }, [id]);
```

Comment:
do we need article id here in the dependency array what happens when we try search for a different article?